crates/hc_run_local_services/src/lib.rs

```diff
@@ -109,30 +109,60 @@ impl HcRunLocalServices {
     }
 
     pub async fn run_err(self) -> Result<()> {
-        let mut task_list = Vec::new();
+        if self.disable_bootstrap && self.disable_signal {
+            println!("All Services Disabled - Aborting");
+            return Ok(());
+        }
 
+        let (bootstrap_shutdown_tx, bootstrap_shutdown_rx) = tokio::sync::oneshot::channel();
         if !self.disable_bootstrap {
             let bs_ip: std::net::IpAddr = self.bootstrap_interface.parse().map_err(Error::other)?;
             let bs_addr = std::net::SocketAddr::from((bs_ip, self.bootstrap_port));
-            let (bs_driver, bs_addr, shutdown) = kitsune_p2p_bootstrap::run(bs_addr, vec![])
-                .await
-                .map_err(Error::other)?;
-            std::mem::forget(shutdown);
-            task_list.push(bs_driver);
-
-            let mut a_out = AOut::new(&self.bootstrap_address_path).await?;
-
-            for addr in tx_addr(bs_addr)? {
-                a_out.write(format!("http://{addr}\n")).await?;
-                println!("# HC BOOTSTRAP - ADDR: http://{addr}");
-            }
-
-            a_out.close().await?;
 
-            println!("# HC BOOTSTRAP - RUNNING");
-        }
-
-        if !self.disable_signal {
+            let mut config = kitsune2_bootstrap_srv::Config::testing();
+            config.listen_address_list = vec![bs_addr];
+
+            std::thread::Builder::new()
+                .name("bootstrap_srv".to_string())
+                .spawn(move || {
+                    // Signal that the bootstrap server is shutting down if the thread dies.
+                    struct D(Option<tokio::sync::oneshot::Sender<()>>);
+                    impl Drop for D {
+                        fn drop(&mut self) {
+                            if let Some(s) = self.0.take() {
+                                let _ = s.send(());
+                            }
+                        }
+                    }
+                    let _d = D(Some(bootstrap_shutdown_tx));
+
+                    let bootstrap_srv = kitsune2_bootstrap_srv::BootstrapSrv::new(config).unwrap();
+
+                    let rt = tokio::runtime::Builder::new_multi_thread()
+                        .enable_io()
+                        .build()
+                        .unwrap();
+                    rt.block_on(async move {
+                        let mut a_out = AOut::new(&self.bootstrap_address_path).await?;
+
+                        for addr in tx_addr(bootstrap_srv.listen_addrs()[0])? {
+                            a_out.write(format!("http://{addr}\n")).await?;
+                            println!("# HC BOOTSTRAP - ADDR: http://{addr}");
```

Comment:
```suggestion
                            tracing::info!("# HC BOOTSTRAP - ADDR: http://{addr}");
```
?

Reply:
We're deliberately writing to stdout here so that people can read the values off the CLI. Maybe not necessary with the file output that's been provided alongside... But this is intentional here. The tracing logs go to stderr and are harder to parse.

Reply:
I see, makes sense to use println then